### PR TITLE
AHU Canonical Type having Four EC Fans

### PR DESCRIPTION
**Pull request by darshanjain12**
"AHU with static pressure-based EC fan speed control, return air temperature-based chilled water valve modulation, filter status monitoring, and four EC fan parameter monitoring."

### CODE REVIEW

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml [hunk 1]
```diff
@@ -4620,6 +4620,19 @@ AHU_BSPC_DX5SC_ECONM_EDM_EFSS_EFVSC_SFSS_SFVSC_SSPC_SARC:
   - SSPC
   - SARC
 
+AHU_SFSS_SFVSC_CHWRC_SSPC_FDPSM_SF4FM::
```
Comment (shambergoldstein) [inline ontology/yaml/resources/HVAC/entity_types/AHU.yaml:4623]: Please clarify - is the set of fans referred to by `SF4FM` separate from the supply fan referred to by `SFSS` and `SFVSC`? Can you explain a bit more about its control?

**Comment by darshanjain12** [comment]
@shambergoldstein Please review

**Comment by shambergoldstein** [comment]
@darshanjain12 please respond to my question from my previous review

**Comment by darshanjain12** [comment]
> @darshanjain12 please respond to my question from my previous review

I Have repliea

> @darshanjain12 please respond to my question from my previous review

It is not separate from SFSS and SFVSC however one AHU comprises of 4 physically separate EC Fan. These fans are speed controlled from single analog output. Each fan is monitored via software integration with following points

supply_fan_energy_accumulator
supply_fan_power_sensor
supply_fan_voltage_sensor
supply_fan_current_sensor
supply_fan_rotationalvelocity_sensor

**Comment by shambergoldstein** [comment]
@darshanjain12 do the fans have individual run statuses?

**Comment by darshanjain12** [comment]
> @darshanjain12 do the fans have individual run statuses?

No We are not having individual fans runstatus

**Comment by darshanjain12** [comment]
@shambergoldstein Request you to review

**Review by shambergoldstein — CHANGES_REQUESTED** [review]

**Comment by darshanjain12** [comment]
@shambergoldstein Done the Required changes as per your comments to keep the model consistent.
Please review

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml [hunk 2]
```diff
@@ -4620,6 +4620,18 @@ AHU_BSPC_DX5SC_ECONM_EDM_EFSS_EFVSC_SFSS_SFVSC_SSPC_SARC:
   - SSPC
   - SARC
 
+AHU_SFSS4X_SFVSC4X_CHWRC_SSPC_FDPSM::
```
Comment (shambergoldstein) [inline ontology/yaml/resources/HVAC/entity_types/AHU.yaml:4623]: As a final change please add a form of economizing.

**Review by shambergoldstein — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml [hunk 3]
```diff
@@ -8454,8 +8470,8 @@ DTPM:
   - detergent_pump_run_status
   - detergent_pump_run_command
   implements:
-  - MONITORING
-
+  - MONITOR
```
Comment (shambergoldstein) [inline ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:8473]: please keep as monitoring

**Review by shambergoldstein — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml (hunk 2 above)
Comment (shambergoldstein) [reply ontology/yaml/resources/HVAC/entity_types/AHU.yaml:4623]: @darshanjain12 please see this request

**Comment by darshanjain12** [comment]
> @darshanjain12 please respond to my question from my previous review

There is no economizer in the system

**Comment by shambergoldstein** [comment]
@darshanjain12 as we expect typical AHUs to have economizing, please either add a form of economizing and mark the fields as missing in the building config, or make this a non-canonical type.

**Comment by darshanjain12** [comment]
> @darshanjain12 as we expect typical AHUs to have economizing, please either add a form of economizing and mark the fields as missing in the building config, or make this a non-canonical type.

@shambergoldstein Done added Economizer

**Comment by shambergoldstein** [comment]
@darshanjain12 please resolve the ontology validator error, after that it should be good to merge

**Comment by darshanjain12** [comment]
> @darshanjain12 please resolve the ontology validator error, after that it should be good to merge

Done Changes

**Review by shambergoldstein — APPROVED** [review]

**Pull request merged**